README.md

```diff
@@ -207,6 +207,94 @@ module.exports = {
 };
 ```
 
+### `encoding`
+
+Type: `String`
+Default: `base64`
+
+Specify the encoding which the file will be in-lined with. It supports [Node.js Buffers and Character Encodings](https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings) which are `["utf8","utf16le","latin1","base64","hex","ascii","binary","ucs2"]`.
+
+**webpack.config.js**
+
+```js
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.svg$/i,
+        use: [
+          {
+            loader: 'url-loader',
+            options: {
+              encoding: 'utf8',
+            },
+          },
+        ],
+      },
+    ],
+  },
+};
+```
+
+### `source`
+
+Type: `Function`
+Default: `data:${mimetype || ''};${encoding},${src.toString(encoding)}`
+
+You can create you own custom implementation for encoding data. `source` argument is a [`Buffer`](https://nodejs.org/api/buffer.html) instance. in the example we are compressing svg files using [mini-svg-data-uri](https://github.com/tigt/mini-svg-data-uri) implementation.
+
+**webpack.config.js**
+
+```js
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.svg$/i,
+        use: [
+          {
+            loader: 'url-loader',
+            options: {
+              source: (svgContentBuffer) => {
+                const svgToMiniDataURI = require('mini-svg-data-uri');
+
+                return svgToMiniDataURI(svgContentBuffer.toString());
+              },
+            },
+          },
+        ],
+      },
+    ],
+  },
+};
+```
+
+By using your own implementation, `mimetype` and `encoding` won't have effect on the final output. until you specify them in the output manually for Example:
+
+```js
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.svg$/i,
+        use: [
+          {
+            loader: 'url-loader',
+            options: {
+              source: (svgContentBuffer) => {
+                return `data:image/svg;utf8,${svgContentBuffer.toString(
```

Comment:
Please add example with `mini-svg-data-uri` package

Reply:
@evilebottnawi I've added an example here: https://github.com/webpack-contrib/url-loader/pull/210/files/2bb2de67e75d175c1ab7b41a075307414f8b6930#diff-04c6e90faac2675aa89e2176d2eec7d8R261 
this example to tell the people what is the final format of the final output to help them develop their own implementation.